agent/tests/test_php_wrapper.c

```diff
@@ -85,6 +90,460 @@ NR_PHP_WRAPPER(test_add_2_arrays) {
+
+  /*
+   * special function for one is called after and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for two is called after and uses NR_OK_TO_OVERWRITE
+   * special function for three is called after and uses NR_NOT_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `two`
+   */
+  setup_nested_framework_calls(NULL, test_name_txn_after_not_ok, NULL,
+                               test_name_txn_after_ok, NULL,
+                               test_name_txn_after_not_ok, "two",
+                               "one:afternotok,two:afterok,three:afternotok");
+
+  /*
+   * special function for one is called after and uses NR_OK_TO_OVERWRITE
+   * special function for two is called after and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for three is called after and uses NR_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `one`
+   */
+  setup_nested_framework_calls(NULL, test_name_txn_after_ok, NULL,
+                               test_name_txn_after_not_ok, NULL,
+                               test_name_txn_after_ok, "one",
+                               "one:afterok,two:afternotok,three:afterok");
+
+  /*
+   * special function for one is called after and uses NR_OK_TO_OVERWRITE
+   * special function for two is called after and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for three is called after and uses NR_NOT_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `one`
+   */
+  setup_nested_framework_calls(NULL, test_name_txn_after_ok, NULL,
+                               test_name_txn_after_not_ok, NULL,
+                               test_name_txn_after_not_ok, "one",
+                               "one:afterok,two:afternotok,three:afternotok");
+
+  /*
+   * special function for one is called after and uses NR_OK_TO_OVERWRITE
+   * special function for two is called after and uses NR_OK_TO_OVERWRITE
+   * special function for three is called after and uses NR_NOT_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `one`
+   */
+  setup_nested_framework_calls(NULL, test_name_txn_after_ok, NULL,
+                               test_name_txn_after_ok, NULL,
+                               test_name_txn_after_not_ok, "one",
+                               "one:afterok,two:afterok,three:afternotok");
+
+  /*
+   * Before only mixes of NR_NOT_OK_TO_OVERWRITE/NR_NOT_OK_TO_OVERWRITE
+   */
+
+  /*
+   * special function for one is called before and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for two is called before and uses NR_OK_TO_OVERWRITE
+   * special function for three is called before and uses NR_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `three`
+   */
+  setup_nested_framework_calls(test_name_txn_before_not_ok, NULL,
+                               test_name_txn_before_ok, NULL,
+                               test_name_txn_before_ok, NULL, "three",
+                               "one:beforenotok,two:beforeok,three:beforeok");
```

Comment:
So the call stack is:
```
before_callback_for_one
  nr_txn_set_path sets txn name as 'one' because it is not set yet
one
  before_callback_for_two
    nr_txn_set_path sets txn name to 'two' because called with NR_OK_TO_OVERWRITE
  two
    before_callback_for_three
      nr_txn_set_path sets txn name to 'three' because called with NR_OK_TO_OVERWRITE
    three
```

Reply:
please see: [857fa16](https://github.com/newrelic/newrelic-php-agent/pull/615/commits/857fa1642e98a49e0a3b46e1404af1b6a4146637)
Convention is to describe what sort of wrapper each function is calling.